networks/3a_validate_array_vs_rnaseq_modules.py

```diff
@@ -0,0 +1,217 @@
+import os
+import scipy
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from sklearn.cluster import DBSCAN, AgglomerativeClustering, AffinityPropagation
+from core_acc_modules import utils, paths
+
+np.random.seed(1)
+
+# +
+# User params
+num_singular_values_log = 100
+
+# Clustering method
+# Choices: {"dbscan", "hierarchal", "affinity"}
+cluster_method = "affinity"
+
+# DBSCAN params
+density_threshold = 8
+
+# Hierarchical clustering params
+hier_threshold = 8
+link_dist = "average"
+
+# Affinity params
+affinity_damping = 0.6
+# -
+
+# Load expression data
+pao1_array_compendium_filename = paths.ARRAY_COMPENDIUM_TO_COMPARE
+pao1_rnaseq_compendium_filename = paths.RNASEQ_COMPENDIUM_TO_COMPARE
+
+pao1_array_compendium = pd.read_csv(
+    pao1_array_compendium_filename, sep="\t", header=0, index_col=0
+)
+pao1_rnaseq_compendium = pd.read_csv(
+    pao1_rnaseq_compendium_filename, sep="\t", header=0, index_col=0
+)
+
+print(pao1_array_compendium.shape)
+pao1_array_compendium.head()
+
+print(pao1_rnaseq_compendium.shape)
+pao1_rnaseq_compendium.head()
+
+# ## Get correlation matrices
+
+# Correlation
+pao1_array_corr_original = pao1_array_compendium.corr()
+
+# Note: Below we plotted the heatmap of the array data to confirm that it has the same issue as the RNA-seq data - there is one large cluster
+
+# %%time
+# Plot heatmap
+"""o1 = sns.clustermap(pao1_array_corr_original.abs(), cmap="viridis", figsize=(20, 20))
+o1.fig.suptitle("Correlation of raw PAO1 genes (array compendium)", y=1.05)"""
+
+# Transpose compendia to be gene x sample
+# Here we're interested in how genes cluster
+pao1_array_compendium_T = pao1_array_compendium.T
+pao1_rnaseq_compendium_T = pao1_rnaseq_compendium.T
+
+# log transform data
+pao1_array_compendium_log10 = np.log10(1 + pao1_array_compendium_T)
+pao1_rnaseq_compendium_log10 = np.log10(1 + pao1_rnaseq_compendium_T)
+
+# Apply SVD
+array_U, array_s, array_Vh = np.linalg.svd(
+    pao1_array_compendium_log10, full_matrices=False
+)
+rnaseq_U, rnaseq_s, rnaseq_Vh = np.linalg.svd(
+    pao1_rnaseq_compendium_log10, full_matrices=False
+)
+
+# Convert ndarray to df to use corr()
+array_U_df = pd.DataFrame(data=array_U, index=pao1_array_compendium_T.index)
+rnaseq_U_df = pd.DataFrame(data=rnaseq_U, index=pao1_rnaseq_compendium_T.index)
+
+# Correlation of U
+# Since `corr()` computes pairwise correlation of columns we need to invert U
+pao1_array_corr_log_spell = array_U_df.iloc[:, :num_singular_values_log].T.corr()
+pao1_rnaseq_corr_log_spell = rnaseq_U_df.iloc[:, :num_singular_values_log].T.corr()
+
+# Note: Here we plot the heatmaps to verify that the correlation of log + SPELL transformed data looks as expected (i.e. there is not a single large cluster)
+
+# Plot heatmap
+"""h1a = sns.clustermap(pao1_array_corr_log_spell.abs(), cmap="viridis", figsize=(20, 20))
+h1a.fig.suptitle(
+    f"log transform + SPELL corrected using {num_singular_values_log} vectors (PAO1 array)",
+    y=1.05,
+)"""
```

Comment:
Just curious: why not comment the code instead of making it a string?

Reply:
This was an easy way to comment multiple lines